server/src/test/java/org/elasticsearch/transport/ClusterConnectionManagerTests.java

```diff
@@ -397,7 +396,7 @@ public void onFailure(Exception e) {
         }
 
         assertTrue(countDownLatch.await(10, TimeUnit.SECONDS));
-        assertTrue(connectCompletionLock.writeLock().tryLock(10, TimeUnit.SECONDS));
+        assertTrue(validatorPermits.tryAcquire(Integer.MAX_VALUE, 10, TimeUnit.SECONDS));
```

Comment:
It would be great to add some message to assertion. May be `Some connections are not released (by validator) in time`?

Reply:
That's what line numbers are for ;) Added in 176a7399561, but 🙏 may we never see them.